src/PhoneNumberInput/PhoneNumberInput.tsx

```diff
@@ -266,15 +268,16 @@ const PhoneNumberInput = forwardRef<HTMLInputElement, Props>((props, ref) => {
           />
         </StyledPhoneNumberWrapper>
         <Flex pointerEvents='auto' position='absolute' right={2}>
-          {icStateIsNotIconToggle() && currentIcRight && (
-            <Icon
-              mr={5}
-              size='small'
-              name={currentIcRight}
-              color={currentIcRightColor}
-              spin={currentIcRight === 'loading'}
-            />
-          )}
+          {icStateIsNotIconToggle() &&
+            (currentIcRight !== 'check' || showValidationCheck) && (
```

Comment:
Extract this boolean condition in a constant as it is quite hard to read and understand what it does

Reply:
9227145d1429dd00450d17d8b7abfaec3edf5c51